### ISSUE TITLE
Verify and proxy options not working together

### ISSUE THREAD

**Issue by archana27m** [guzzle/guzzle#1861]
I am facing issue using 'verify' setting to CA file and setting up proxy with http & https options.

$result = $client->request ( 'POST', /, [ 
					'verify' => 'C:\xampp\cacart.pem',
					'proxy' => [
				'http'  => '127.0.0.1:8080', // Use this proxy with "http"
				'https' => '127.0.0.1:8080' // Use this proxy with "https"
			 ]);
		
Getting error as "cURL error 60: SSL certificate problem: self signed certificate in certificate chain (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)"

Please let me know, any possible solution to this.

**Comment by sagikazarmark** [comment]
Are you sure your proxy supports TLS handshake tunneling? Not all proxies support that.

First hit on google: https://security.stackexchange.com/questions/115762/how-is-it-possible-to-do-tls-through-proxy-without-anyone-noticing

**Comment by archana27m** [comment]
@sagikazarmark, I am not sure if it supports TLS but If I set 'Verify' to 'False' then it works with HTTPS
otherwise it works with HTTP only.

**Comment by sagikazarmark** [comment]
Just a hunch, but shouldn't `C:\xampp\cacart.pem` be `C:\xampp\cacert.pem` ? (Not sure which one is present on your system)

The error says that the certificate seems to be a self-signed one meaning it cannot be verified with any of the CA certs .... maybe because Guzzle doesn't see any?

**Comment by archana27m** [comment]
@sagikazarmark  Sorry, that was typo its cacert.pem file, I downloaded the latest one from the site. and using it to specify ssl verification for guzzle. I am using burp suite for proxy and testing it locally. 

When I set 'verify' to 'false' it works with proxy - http and https as well. (Setting 'verify' to 'false' is not preferable). 

It would be helpful if I could get it working with cert file and proxy with http and https
